src/main/java/com/kodilla/checkers/io/BoardGame.java

```diff
@@ -0,0 +1,316 @@
+package com.kodilla.checkers.io;
+
+import com.kodilla.checkers.logic.Board;
+import com.kodilla.checkers.logic.BoardPosition;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.paint.Color;
+
+import java.util.*;
+
+public class BoardGame {
+
+    private final double pieceMargin;
+    private final double startX;
+    private final double startY;
+    private final double sideLength;
+    private final double unitLength;
+    private Board board;
+    private List<BoardPosition> legalPos;
+    boolean lastColor;
+    boolean gameOver;
+    boolean opponentSet;
+
+    public BoardGame(double startX, double startY, double sideLength, double pieceMargin, int sideCount, int startCount) {
```

Comment:
Jak masz taki "zbiór" parametrów o podobnym przeznaczeniu (tutaj wszystkie 6 parametrów opisuje jakieś właściwości wizualne), to warto wrzucić to do jednej klasy (np. `BoardVisualSettings`), przekazywać tylko ten jeden obiekt jako argument. Dodatkowy plus - będziesz miał o 5 pól mniej w `BoardGame`, znacznie łatwiej będzie to czytać :)